xonsh/built_ins.py

```diff
@@ -180,7 +182,9 @@ def expand_path(s):
     global ENV
     if ENV is not None:
         ENV.replace_env()
-    return os.path.expanduser(os.path.expandvars(s))
+    if ENV.get('EXPAND_ENV_VARS'):
+        s = expandvars(s)
```

Comment:
Are `replace_env` and `undo_replace_env` used someplace else I am not seeing?  I am planning on just deleting them unless they are useful someplace else.